aten/src/ATen/native/cuda/Distributions.cu

```diff
@@ -441,11 +443,77 @@ void uniform_kernel_cuda(TensorIterator& iter, double from_, double to_, Generat
    });
 }
 
+void random_kernel_cuda(TensorIterator& iter, uint64_t range, int64_t base, Generator* gen_) {
+  auto gen = check_generator<CUDAGenerator>(gen_, &globalContext().defaultGenerator(kCUDA));
+  AT_DISPATCH_ALL_TYPES_AND(at::ScalarType::Half, iter.dtype(), "random_cuda", [&] {
+    using accscalar_t = at::acc_type<scalar_t, true>;
+    if (std::is_same<scalar_t, double>::value || std::is_same<scalar_t, int64_t>::value) {
+      // define lambda to mod with range and add base
+      auto random_func = [range, base] __device__ (uint64_t rand) {
+        return static_cast<scalar_t>(rand % range + base);
```

Comment:
Heh heh, modulo isn't uniform, but we're unlikely to switch this to rejection sampling. No changes needed, but it did look funny to see :>

Reply:
Oh yeah :). Will leave that discussion for another day.

Reply:
There are 2 orthogonal problems with modulo
1) when `range` is comparable to the range produced by generator, modulo indeed produces non-uniform results. Usually not a problem though, because in typical usecases `range` is much less than range of random numbers produced
2) modulo essentially extracts lower bits, and lower bits have low entropy. That's true for lcg's, but we are using philox here, so should be fine.